File: src/main/java/org/taktik/icure/services/external/rest/v1/mapper/utils/InstantMapper.java
/*
 * Copyright (c) 2020. Taktik SA, All rights reserved.
 */

package org.taktik.icure.services.external.rest.v1.mapper.utils;

import org.mapstruct.InjectionStrategy;
import org.mapstruct.Mapper;
import java.time.Instant;

@Mapper(componentModel = "spring", injectionStrategy = InjectionStrategy.CONSTRUCTOR)
public abstract class InstantMapper {
    Long map(Instant instant) { return instant.toEpochMilli(); }
    Instant map(Long aLong) { return Instant.ofEpochMilli(aLong); }
}

File: src/main/kotlin/org/taktik/icure/services/external/rest/v1/mapper/utils/InstantMapper.kt
/*
 * Copyright (c) 2020. Taktik SA, All rights reserved.
 */
package org.taktik.icure.services.external.rest.v1.mapper.utils

import java.time.Instant
import org.mapstruct.InjectionStrategy
import org.mapstruct.Mapper

@Mapper(componentModel = "spring", injectionStrategy = InjectionStrategy.CONSTRUCTOR)
abstract class InstantMapper {
	fun map(instant: Instant): Long {
		return instant.toEpochMilli()
	}

	fun map(aLong: Long?): Instant {
		return Instant.ofEpochMilli(aLong!!)
	}
}
